### ISSUE TITLE
[a11y]: ComposedModal [all variants] - identifies visually hidden controls

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13888]
### Package

@carbon/react

### Browser

Safari & FF

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology

JAWS & VO

### Description

ComposedModal [all variants]
Screen reader identifies visually hidden controls at the beginning and the end of the modal

### WCAG 2.1 Violation

4.1.2 Name, Role, Value 

### Reproduction/example

Record `Number 24` in the below table: 
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce

In JAWS with FF, After the dialog is displayed, When we navigate with the up/down arrow keys through the dialog content, Screen reader identifies A button with the label "Focus sentinel" at the beginning and end of the dialog. This is visually not present and identified only by the screen reader.

In VO iPhone Safari, The same issue exists. Along with that, When We navigate to the "Focus sentinel" button at the beginning of the dialog, VO focus wraps back to the "Add" button automatically. Similarly, when we navigate to the "Focus sentinel" button in the end, the VO focus wraps back to the "close" button at the top.

### User impact: 
When screen reader identifies the visually hidden elements, it can be misleading for the blind user

### Recommended Fix: 
The mentioned controls should be hidden from the screen reader as well

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
